src/ristretto/pedersen/extended_commitment_factory.rs

```diff
@@ -0,0 +1,697 @@
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use bulletproofs_plus::{generators::pedersen_gens::ExtensionDegree, PedersenGens};
+use curve25519_dalek::{ristretto::RistrettoPoint, traits::Identity};
+
+use crate::{
+    commitment::{ExtendedHomomorphicCommitmentFactory, HomomorphicCommitment, HomomorphicCommitmentFactory},
+    errors::CommitmentError,
+    ristretto::{
+        constants::{RISTRETTO_NUMS_POINTS, RISTRETTO_NUMS_POINTS_COMPRESSED},
+        pedersen::{
+            PedersenCommitment,
+            RISTRETTO_PEDERSEN_G,
+            RISTRETTO_PEDERSEN_G_COMPRESSED,
+            RISTRETTO_PEDERSEN_H,
+            RISTRETTO_PEDERSEN_H_COMPRESSED,
+        },
+        RistrettoPublicKey,
+        RistrettoSecretKey,
+    },
+};
+
+/// Generates extended Pederson commitments `sum(k_i.G_i) + v.H` using the provided base
+/// [RistrettoPoints](curve25519_dalek::ristretto::RistrettoPoints).
+/// Notes:
+///  - Homomorphism with public key only holds for extended commitments with `ExtensionDegree::Zero`
+#[derive(Debug, PartialEq, Clone)]
+pub struct ExtendedPedersenCommitmentFactory(pub(crate) PedersenGens<RistrettoPoint>);
+
+impl ExtendedPedersenCommitmentFactory {
+    /// Create a new Extended Pedersen Ristretto Commitment factory for the required extension degree using
+    /// pre-calculated compressed constants - we only hold references to the static generator points.
+    pub fn new_with_extension_degree(extension_degree: ExtensionDegree) -> Result<Self, CommitmentError> {
```

Comment:
I'm not too fond of the use of ExtensionDegree here. The type is in the `bulletproofs_plus` crate, meaning that you can't use this crate without Bulletproofs, for example, if it were hidden behind a feature flag. `PedersonGens` is perhaps in the wrong crate, or perhaps we should use a field in the `ExtendedPedersonCommitmentFactory` `gens: Vec<RistrettoPoint>`

Reply:
As we are dependent on 
```
use bulletproofs_plus::{generators::pedersen_gens::ExtensionDegree, PedersenGens};
```
here and if we need to remove the dependency I think the most elegant solution would be to completely move `generators` out of `bulletproofs_plus` into its own crate.

If that solution is acceptable I propose we deal with that as a separate series of PRs.

Reply:
Good point. Would it be better to rewrite `ExtensionDegree` as a newtype around u64 rather than an enum, since the code in bp+ always converts it to an int anyway. 
You could then add an adapter type in tari_crypto to resolve the dependency issue.